packages/web-ui/src/sections/Document/Editor/Playground/Chat.tsx

```diff
@@ -0,0 +1,217 @@
+
+  const [conversation, setConversation] = useState<Conversation | undefined>(
+    undefined,
+  )
+
+  const [responseStream, setResponseStream] = useState<string | undefined>(
+    undefined,
+  )
+  const StreamMessage = useMemo(() => {
+    if (responseStream === undefined) return null
+    if (conversation === undefined) return null
+    if (conversation.messages.length < chainLength - 1)
+      return (
+        <Message
+          role={MessageRole.assistant}
+          content={[{ type: ContentType.text, value: responseStream }]}
+          className='animate-pulse'
+        />
+      )
+    if (conversation.messages.length === chainLength - 1)
+      return (
+        <Message
+          role={MessageRole.assistant}
+          content={[{ type: ContentType.text, value: responseStream }]}
+          badgeVariant='accent'
+          textColor='accentForeground'
+        />
+      )
+    return (
+      <Message
+        role={MessageRole.assistant}
+        content={[{ type: ContentType.text, value: responseStream }]}
+        badgeVariant='outline'
+      />
+    )
+  }, [responseStream, conversation, chainLength])
+
+  const addMessage = useCallback((message: ConversationMessage) => {
+    let newConversation: Conversation
+    setConversation((prevConversation) => {
+      newConversation = {
+        ...prevConversation,
+        messages: [...(prevConversation?.messages ?? []), message],
+      } as Conversation
+      return newConversation
+    })
+    return newConversation!
+  }, [])
+
+  const generateResponse = useCallback(async (_: Conversation) => {
+    const mockResponse = faker.hacker.phrase()
+    let response = ''
+    setResponseStream(response)
+
+    for await (const [char, tokenCount] of mockAIResponse(mockResponse)) {
+      response += char
+      setResponseStream(response)
+      setTokens((prev) => prev + tokenCount)
+    }
+
+    setResponseStream(undefined)
+    addMessage({
+      role: MessageRole.assistant,
+      content: [
+        {
+          type: ContentType.text,
+          value: response,
+        },
+      ],
+      toolCalls: [],
+    })
+    return response
+  }, [])
+
+  const runChain = useCallback((lastResponse?: string) => {
+    chain
+      .step(lastResponse)
+      .then(async ({ completed, conversation }) => {
+        if (completed) setChainLength(conversation.messages.length + 1)
+        setConversation(conversation)
+
+        const response = await generateResponse(conversation)
+
+        if (completed) return
+        runChain(response)
+      })
+      .catch((error) => {
+        setError(error)
+        if (error instanceof CompileError) {
+          console.error(error.toString())
+        } else {
+          console.log(error)
+        }
+      })
+  }, [])
+
+  useEffect(() => {
+    if (runChainOnce.current) return
+    runChainOnce.current = true // Prevent double-running when StrictMode is enabled
+    runChain()
+  }, [])
```

Comment:
When Playground enters mode `chat`, it starts the chain generation right when it is rendered.